src/divide.rs

```diff
@@ -37,16 +39,22 @@ use std::thread;
 /// # assert_eq!(v, w);
 /// # }
 /// ```
-pub fn divide<T, F>(data: &mut [T], granularity: usize, operation: F) where
+pub fn divide<T, F>(data: &mut [T], granularity: Option<usize>, operation: F) where
     T: Send,
     F: Fn(&mut [T], usize) + Sync,
 {
-    assert!(granularity > 0);
+    let true_granularity : usize = match granularity {
+        None => max(data.len()/std::os::num_cpus(), 1),
```

Comment:
Could you use `data.len() / num_cpus + 1` here? That way this should never spawn more than `num_cpus` threads.
(In it's current state, if `data` has 15 elements, and `num_cpus = 8`, this will spawn 15 threads)